docs/training/apis-howto/api-core-ug.rst

```diff
@@ -301,110 +457,145 @@ considerations are:
+   device = torch.device(core_context.distributed.local_rank)
 
-#. Create a ``4_distributed.yaml`` file by copying the ``3_distributed.yaml`` file and changing the
-   first couple of lines:
+Shard the data into ``num_replicas`` non-overlapping parts. ``num_replicas`` is equal to
+``core_context.distributed.size``, or the number of slots:
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/4_distributed.yaml
-      :language: yaml
-      :lines: 1-2
+.. code:: python
 
-   Set the ``resources.slots_per_trial`` field to the number of GPUs you want:
+   # NEW: Create DistributedSampler object for sharding data into
+   # core_context.distributed.size parts.
+   sampler1 = DistributedSampler(
+       dataset1,
+       num_replicas=core_context.distributed.size,
+       rank=core_context.distributed.rank,
+       shuffle=True,
+   )
+   sampler2 = DistributedSampler(
+       dataset2,
+       num_replicas=core_context.distributed.size,
+       rank=core_context.distributed.rank,
+       shuffle=True,
+   )
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/4_distributed.yaml
-      :language: yaml
-      :start-at: resources:
-      :end-at: slots_per_trial:
+   # NEW: Shard data.
+   train_loader = torch.utils.data.DataLoader(dataset1, sampler=sampler1, **train_kwargs)
+   test_loader = torch.utils.data.DataLoader(dataset2, sampler=sampler2, **test_kwargs)
+
+Wrap your model with torch’s DistributedDataParallel:
+
+.. code:: python
+
+   # NEW: Wrap model with DDP. Aggregates gradients and
+   # synchronizes model training across slots.
+   model = DDP(model, device_ids=[device], output_device=device)
+
+Finally, at each place in the code where you upload checkpoints, report training metrics, or report
+progress to the master, make sure this is done only on rank 0, e.g.,:
+
+.. code::
+
+   # NEW: Report metrics only on rank 0: only the chief worker
+   # may report training metrics and progress, or upload checkpoints.
+   if core_context.distributed.rank == 0:
+       core_context.train.report_training_metrics(
+           steps_completed=(batch_idx + 1) + (epoch - 1) * len(train_loader),
+           metrics={“train_loss”: loss.item()},
+       )
+
+Step 5.3: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code:: bash
 
-   You can return to using the ``single`` searcher instead of an ``adaptive_asha`` hyperparameter
-   search:
+   det e create distributed.yaml .
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/4_distributed.yaml
-      :language: yaml
-      :start-at: searcher:
-      :end-at: max_length:
+In the Determined WebUI, visit the **Cluster** tab.
 
-#. Run the code using the Determined CLI with the following command:
+You should be able to see multiple slots active in the Cluster tab corresponding to the value you
+set for slots_per_trial you set in distributed.yaml, as well as logs appearing from multiple ranks.
```

Comment:
suggest double-backticks for `slots_per_trial` and `distributed.yaml`

Reply:
resolved and found a few more